File: code/151.reverse-words-in-a-string.php
<?php
/*
 * @lc app=leetcode id=151 lang=php
 *
 * [151] Reverse Words in a String
 */

// @lc code=start
class Solution {

    /**
     * @param String $s
     * @return String
     */
    function reverseWords($s) {
        $result = "";
        foreach (array_reverse(explode(" ", trim($s))) as $value) {
            if (trim($value) !== "" ) {
                $result .= $value. ' ';
            }
        }
        return rtrim($result);
    }
}
// @lc code=end

File: code/151.reverse-words-in-a-string.go
package code

import (
	"slices"
	"strings"
)

/*
 * @lc app=leetcode id=151 lang=golang
 *
 * [151] Reverse Words in a String
 */

// @lc code=start
func reverseWords(s string) string {
	s = strings.TrimSpace(s)
	a := strings.Fields(s)
	slices.Reverse(a)

	return strings.Join(a, " ")
}

// @lc code=end
